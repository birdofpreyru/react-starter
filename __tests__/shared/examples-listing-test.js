/**
 * @jest-environment @dr.pogodin/react-utils/jest-e2e-ssr-env
 * @webpack-config-factory ./config/webpack/development.js
 * @webpack-config-options {
 *  "entry": "./__assets__/index.js",
 *  "dontUseHmr": true,
 *  "dontUseProgressPlugin": true,
 *  "dontUseReactGlobalStateDebugging": true }
 * @ssr-options { "entry": "./__assets__/Scene.jsx" }
 */

// This test both tests that the examples listing is correctly rendered both in
// SSR and client-side, and it also tests that E2eSsrEnv Jest environment works
// fine outside the react-utils.

// eslint-disable-next-line import/no-extraneous-dependencies
import { act } from 'react-dom/test-utils';

it('performs correct SSR and client-side hydration', async () => {
  document.write(global.ssrMarkup);
  const view = document.querySelector('#react-view').innerHTML;
  expect(view).toMatchSnapshot();
  const outputPath = global.webpackConfig.output.path;
  const js = global.webpackOutputFs.readFileSync(
    `${outputPath}/main.js`,
    'utf8',
  );
  await act(new Function(js)); // eslint-disable-line no-new-func
  expect(document.querySelector('#react-view').innerHTML).toBe(view);
});
