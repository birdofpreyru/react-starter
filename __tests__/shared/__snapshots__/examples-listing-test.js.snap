// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs correct SSR and client-side hydration 1`] = `"<!--$--><div class="-dr-pogodin-react-utils___src-shared-components-PageLayout-base-theme___container___T3cuHB"><div class="-dr-pogodin-react-utils___src-shared-components-PageLayout-base-theme___sidePanel___lqNh4h "></div><div class="-dr-pogodin-react-utils___src-shared-components-PageLayout-base-theme___mainPanel___pPlQO2"><h1>Dr. Pogodin's React Starter</h1><ul><li><a href="/api-client">API (HTTP) client</a></li><li><a href="/async-global-data">Async Global Data</a></li><li><a href="/code-splitting">Code Splitting</a></li><li><a href="/buttons">Buttons</a></li><li><a href="/global-state">Global State</a></li><li><a href="/input-form-components">Input Form Components</a></li><li><a href="/links">Link Tests</a></li><li><a href="/modal">Modal</a></li><li><a href="/page-layout">Page layout</a></li><li><a href="/react-themes">React themes</a></li><li><a href="/svg">SVG Demo</a></li><li><a href="/throbber">Throbber (activity indicator)</a></li><li><a href="/tooltip">Tooltip</a></li><li><a href="/youtube-video">YouTube video</a></li></ul></div><div class="-dr-pogodin-react-utils___src-shared-components-PageLayout-base-theme___sidePanel___lqNh4h "></div></div><!--/$-->"`;
